tools/xtce_generation/telemetry_generator.py

```diff
@@ -125,29 +91,26 @@ def create_parameter_types(
         Parameters:
         - parameter_type_set: The ParameterTypeSet element where parameter types are.
         - unique_lengths: Unique values from the 'lengthInBits' column.
-        - data_type_data: The data type for the "Data" mnemonic.
-        - data_type_event_data: The data type for the "Event_Data" mnemonic.
-        - sci_byte: The BYTE number of lengthInBits in both "Data" and "Event_Data"
+        - data_types: a dictionary mapping the given mnemonic names to the
+        data types from the packet.
 
         Returns:
         - parameter_type_set: The updated ParameterTypeSet element.
         """
         for size in unique_lengths:
-            parameter_type_ref = None
+            parameter_type_ref = f"uint{size}"  # Use UINT for other sizes
 
-            if size == sci_byte:
+            if size == self.sci_byte:
                 parameter_type_ref = (
                     "BYTE"  # Use BYTE type for both "Data" and "Event_Data"
                 )
             elif size == self.sci_byte:
-                parameter_type_ref = (
-                    data_type_data  # Use data_type_data for "Data" mnemonic
-                )
-            elif size == self.sci_byte:
+                parameter_type_ref = data_types[
+                    "Data"
+                ]  # use dataType of "Data" in data_types dict
+            elif data_types["EventData"] and size == self.sci_byte:
```

Comment:
is "EventData" a CoDICE specific thing? I'm getting an error because I don't have "EventData" defined as a key

Reply:
Yes it is codice specific. But you can change the mnemonic. I also think Maxine might have fixed this, so when pushed, it should work..... If this helps, I have packets with "Data" OR "Event_Data" so it was my way of calling the correct one depending on what was in the row. 

 - data_type_event_data: The data type for the "Event_Data" mnemonic

Reply:
Ah yes, it is codice specific. In my GLOWs code, I override it: 

```    def create_parameter_types(
        self,
        parameter_type_set,
        unique_lengths,
        data_types,
        sci_byte=None,
    ):
        """
        Create parameter types based on 'dataType' for the unique 'lengthInBits' values.
        This will loop through the unique lengths and create a ParameterType element
        for each length representing a UINT type.

        Parameters:
        - parameter_type_set: The ParameterTypeSet element where parameter types are.
        - unique_lengths: Unique values from the 'lengthInBits' column.
        - data_type_data: The data type for the "Data" mnemonic.
        - data_type_event_data: The data type for the "Event_Data" mnemonic.
        - sci_byte: The BYTE number of lengthInBits in both "Data" and "Event_Data"

        Returns:
        - parameter_type_set: The updated ParameterTypeSet element.
        """
        for size in unique_lengths:
            # All bins equal?
            # Are bins the only thing that matters? -> check glows code for this
            parameter_type_ref = data_types["BIN"]  # Use UINT for other sizes
            print(parameter_type_ref)
            parameter_type = Et.SubElement(parameter_type_set, "xtce:ParameterType")
            parameter_type.attrib["name"] = f"uint{size}"
            parameter_type.attrib["signed"] = "false"

            encoding = Et.SubElement(parameter_type, "xtce:IntegerDataEncoding")
            encoding.attrib["sizeInBits"] = str(size)
            encoding.attrib["encoding"] = "unsigned"

            # Create BinaryParameterType if parameter_type_ref is "BYTE"
            if parameter_type_ref == "BYTE":
                binary_parameter_type = Et.SubElement(
                    parameter_type_set, "xtce:BinaryParameterType"
                )
                binary_parameter_type.attrib["name"] = parameter_type_ref

                Et.SubElement(binary_parameter_type, "xtce:UnitSet")

                binary_data_encoding = Et.SubElement(
                    binary_parameter_type, "xtce:BinaryDataEncoding"
                )
                binary_data_encoding.attrib["bitOrder"] = "mostSignificantBitFirst"

                size_in_bits = Et.SubElement(binary_data_encoding, "xtce:SizeInBits")
                fixed_value = Et.SubElement(size_in_bits, "xtce:FixedValue")
                fixed_value.text = str(sci_byte)  # Set the size in bits to sci_byte

        return parameter_type_set
```

Reply:
Ideally, in the future, this will be something that is probably overridden in most instruments, or the more generic code above can be modified so it can be used in all cases where there are very simple parameters.